src/psyclone/psyGen.py

```diff
@@ -5768,7 +5825,48 @@ def _part_ref_handler(self, node, parent):
         '''
         from fparser.two import Fortran2003
 
-        reference_name = node.items[0].string
+        reference_name = node.items[0].string.lower()
+
+        # Intrinsics are wrongly parsed as arrays by fparser2 (fparser issue
+        # #189), we can fix the issue here and convert them to appropiate PSyIR
+        # nodes.
+        if reference_name == 'sign':
+            bop = BinaryOperation(BinaryOperation.Operator.SIGN, parent)
+            self.process_nodes(parent=bop, nodes=[node.items[1].items[0]],
+                               nodes_parent=node)
+            self.process_nodes(parent=bop, nodes=[node.items[1].items[1]],
+                               nodes_parent=node)
+            return bop
+        elif reference_name == 'sin':
+            uop = UnaryOperation(UnaryOperation.Operator.SIN, parent)
+            self.process_nodes(parent=uop, nodes=[node.items[1]],
+                               nodes_parent=node)
+            return uop
+        elif reference_name == 'real':
+            if isinstance(node.items[1], Fortran2003.Section_Subscript_List):
```

Comment:
I added a `if len(node.items) != 2:` test as I am expecting fparser2 nodes to look like:`Part_Ref(Name('real'), Name('a'))`.

The second items can subsequently have more items (e.g. in `real(a+b)`).

Is this what you mean?

Reply:
No, that wasn't quite it :-) I meant that you're using the fact that `node.items[1]` is an instance of `Section_Subscript_List` to indicate that it has more than a single argument and thus you create a CodeBlock. In future I think the plan (https://github.com/stfc/fparser/issues/170)  is that `node.items[1]` will *always* be a `Section_Subscript_List`, only it may contain just a single entry.

Reply:
OK, I think I got it now.

Anyway, if fparser2 support for Intrinsic is introduced maybe we can simplify this altogether.

Reply:
Oh yes, good point.